internal/packager/fork.go

```diff
@@ -0,0 +1,118 @@
+package packager
+
+import (
+	"archive/tar"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/docker/app/internal"
+	"github.com/docker/app/internal/types"
+
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+	yaml "gopkg.in/yaml.v2"
+)
+
+// Fork pulls an application and creates a local fork for the user to modify
+func Fork(originName, forkName, outputDir string, maintainers []string) error {
```

Comment:
Lot's of what is done here is already done elsewhere in `packager` and other packages… we should factor code at some point…

Reply:
Do you have specific parts of the function in mind? I tried to reuse existing code as much as possible I think. Except maybe the tar handling?

Reply:
Tar handling, temporary extract/load/… yes

Reply:
Created #325 for this so it doesn't block this PR

Reply:
Having #325 not depending on this PR would have been either for the review :angel: